modules/remotecontrol/Receiver.js

```diff
@@ -1,331 +0,0 @@
-
-import RemoteControlParticipant from './RemoteControlParticipant';
-
-declare var APP: Object;
-declare var config: Object;
-declare var interfaceConfig: Object;
-
-const logger = getLogger(__filename);
-
-/**
- * The transport instance used for communication with external apps.
- *
- * @type {Transport}
- */
-const transport = new Transport({
-    backend: new PostMessageTransportBackend({
-        postisOptions: { scope: 'jitsi-remote-control' }
-    })
-});
-
-/**
- * This class represents the receiver party for a remote controller session.
- * It handles "remote-control-event" events and sends them to the
- * API module. From there the events can be received from wrapper application
- * and executed.
- */
-export default class Receiver extends RemoteControlParticipant {
-    _controller: ?string;
-    _enabled: boolean;
-    _hangupListener: Function;
-    _remoteControlEventsListener: Function;
-    _userLeftListener: Function;
-
-    /**
-     * Creates new instance.
-     */
-    constructor() {
-        super();
-        this._controller = null;
-        this._remoteControlEventsListener
-            = this._onRemoteControlMessage.bind(this);
-        this._userLeftListener = this._onUserLeft.bind(this);
-        this._hangupListener = this._onHangup.bind(this);
-
-        // We expect here that even if we receive the supported event earlier
-        // it will be cached and we'll receive it.
-        transport.on('event', event => {
-            if (event.name === REMOTE_CONTROL_MESSAGE_NAME) {
-                this._onRemoteControlAPIEvent(event);
-
-                return true;
-            }
-
-            return false;
-        });
-    }
-
-    /**
-     * Enables / Disables the remote control.
-     *
-     * @param {boolean} enabled - The new state.
-     * @returns {void}
-     */
-    _enable(enabled: boolean) {
-        if (this._enabled === enabled) {
-            return;
-        }
-        this._enabled = enabled;
-        if (enabled === true) {
-            logger.log('Remote control receiver enabled.');
-
-            // Announce remote control support.
-            APP.connection.addFeature(DISCO_REMOTE_CONTROL_FEATURE, true);
-            APP.conference.addConferenceListener(
-                JitsiConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
-                this._remoteControlEventsListener);
-            APP.conference.addListener(JitsiMeetConferenceEvents.BEFORE_HANGUP,
-                this._hangupListener);
-        } else {
-            logger.log('Remote control receiver disabled.');
-            this._stop(true);
-            APP.connection.removeFeature(DISCO_REMOTE_CONTROL_FEATURE);
-            APP.conference.removeConferenceListener(
-                JitsiConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
-                this._remoteControlEventsListener);
-            APP.conference.removeListener(
-                JitsiMeetConferenceEvents.BEFORE_HANGUP,
-                this._hangupListener);
-        }
```

Comment:
I don't see this addFeature and removeFeature in the PR, where are those?
Is there a case where the remote control can be disabled live, during a session?

Reply:
This event was emitted by `hangup()` from conference.js. Now we directly dispatch `disableReceiver()` from there.

Reply:
I can't think of a case when the the remote control will be disabled live. If such an use case appears we can add some logic to handle it. But currently the new code should be equivalent to the old one.